ext/dom/php_dom.c

```diff
@@ -421,7 +421,7 @@ static HashTable* dom_get_debug_info_helper(zend_object *object, int *is_temp) /
 	ZEND_HASH_MAP_FOREACH_STR_KEY_PTR(prop_handlers, string_key, entry) {
 		zval value;
 
-		if (entry->read_func(obj, &value) == FAILURE || !string_key) {
+		if (entry->read_func(obj, &value) == FAILURE) {
```

Comment:
Make it an assertion that the string_key exists? IIRC there was a "bug" when arrays with numeric strings where casted to objects and thus had int property names.

It shouldn't happen anymore but an assertion is cheap...